src/main/scala/firrtl/passes/Checks.scala

```diff
@@ -155,47 +156,84 @@ object CheckHighForm extends Pass {
       }
       e foreach checkHighFormW(info, mname)
       e foreach checkHighFormT(info, mname)
-      e foreach checkHighFormE(info, mname, names)
+      e foreach checkHighFormE(info, mname, names, literals)
     }
 
     def checkName(info: Info, mname: String, names: NameSet)(name: String): Unit = {
       if (names(name))
         errors.append(new NotUniqueException(info, mname, name))
       names += name
     }
+    
+    def resolveToLiteral(literals: LiteralSet, e: Expression): Boolean = {
+      if(e.isInstanceOf[Literal])
+        return true
+      e match {
+        case t: HasName => literals.contains(t.name) && literals(t.name)
+        case t: WRef => literals.contains(t.name) && literals(t.name)
+        case t: WSubIndex => resolveToLiteral(literals, t.expr)
+        case t: WSubField => resolveToLiteral(literals, t.expr)
+        case t: WSubAccess => resolveToLiteral(literals, t.expr)
+        case _ => false
+      }
+    }
```

Comment:
I think this would be better expressed as:
```scala
     def resolveToLiteral(literals: LiteralSet, e: Expression): Boolean = {
        case _: Literal => true
        case t: HasName => literals.contains(t.name) && literals(t.name)
        case t: WRef => literals.contains(t.name) && literals(t.name)
        case t: WSubIndex => resolveToLiteral(literals, t.expr)
        case t: WSubField => resolveToLiteral(literals, t.expr)
        case t: WSubAccess => resolveToLiteral(literals, t.expr)
        case _ => false
      }
    }